loki/transformations/single_column/annotate.py

```diff
@@ -69,19 +69,40 @@ def annotate_vector_loops(self, routine):
-                for pragma in as_tuple(loop.pragma):
-                    if is_loki_pragma(pragma, starts_with='loop vector reduction'):
-                        # Turn reduction pragmas into `!$acc` equivalent
-                        pragma._update(keyword='acc')
-                        continue
-
-                    if is_loki_pragma(pragma, starts_with='loop vector'):
-                        # Turn general vector pragmas into `!$acc` and add private clause
-                        private_arrs = ', '.join(v.name for v in private_arrays)
-                        private_clause = '' if not private_arrays else f' private({private_arrs})'
-                        pragma._update(keyword='acc', content=f'loop vector{private_clause}')
+        if self.directive == 'openacc':
+            with pragmas_attached(routine, ir.Loop):
+                for loop in FindNodes(ir.Loop).visit(routine.body):
+                    for pragma in as_tuple(loop.pragma):
+                        if is_loki_pragma(pragma, starts_with='loop vector reduction'):
+                            # Turn reduction pragmas into `!$acc` equivalent
+                            pragma._update(keyword='acc')
+                            continue
+
+                        if is_loki_pragma(pragma, starts_with='loop vector'):
+                            # Turn general vector pragmas into `!$acc` and add private clause
+                            private_arrs = ', '.join(v.name for v in private_arrays)
+                            private_clause = '' if not private_arrays else f' private({private_arrs})'
+                            pragma._update(keyword='acc', content=f'loop vector{private_clause}')
+
+        if self.directive == 'omp-gpu':
+            with pragmas_attached(routine, ir.Loop):
+                for loop in FindNodes(ir.Loop).visit(routine.body):
+                    for pragma in as_tuple(loop.pragma):
+                        #TODO: how to handle vector reductions?
+
+                        if is_loki_pragma(pragma, starts_with='loop vector'):
+                            # TODO: need for privatizing variables/arrays?
+                            pragma_new = ir.Pragma(keyword='omp', content='parallel do')
+                            pragma_post = ir.Pragma(keyword='omp', content='end parallel do')
+                            # pragma_new = ir.Pragma(keyword='omp', content='loop bind(parallel)')
+                            # pragma_post = ir.Pragma(keyword='omp', content='end loop')
+
+                            # Replace existing loki pragma and add post-pragma
+                            loop_pragmas = tuple(p for p in as_tuple(loop.pragma) if p is not pragma)
+                            loop._update(
+                                pragma=loop_pragmas + (pragma_new,),
+                                pragma_post=(pragma_post,) + as_tuple(loop.pragma_post)
+                            )
```

Comment:
Same comment, essentially: Encode the parallelisation concepts used here as a generic interface that applies them, and then overload as programming model-specific implementations for OpenMP and OpenACC.

I'm undecided whether this should be captured together with the data offload interfaces in common programming-model specific classes or have them separate by function (i.e., parallelization, data movement).